### PR TITLE
Automate cross-compilation and release of JNI libs for Android

### LINKED ISSUE #1608: Automate cross-compilation and release of JNI libs for Android

### PR DESCRIPTION
**Pull request by samuel-rufi**
Fixes https://github.com/iotaledger/wallet.rs/issues/1608

### CODE REVIEW

File: .github/workflows/java_binding_publish.yml [hunk 1]
```diff
@@ -128,4 +136,117 @@ jobs:
           ./gradlew publishToSonatype
           if [ "$RUNNER_OS" == "macOS" ];
           then ./gradlew publishToSonatype -PbuildTarget=aarch64-apple-darwin
-          fi
\ No newline at end of file
+          fi
+
+  release-android:
+    name: Create Android releases
+    needs: release-desktop
```
Comment (Thoralf-M) [inline .github/workflows/java_binding_publish.yml]: Is this just for `needs.release-desktop.outputs.tag`?

Comment (samuel-rufi) [reply .github/workflows/java_binding_publish.yml]: yes

Comment (Thoralf-M) [reply .github/workflows/java_binding_publish.yml]: The version is the one defined in build.gradle? Would be nice to get it directly from there, then you don't need to extract it from the jar file name and you also don't need to wait for the first build to finish before you start the android one

Comment (samuel-rufi) [reply .github/workflows/java_binding_publish.yml]: good idea, will try something

File: .github/workflows/java_binding_publish.yml [hunk 2]
```diff
@@ -128,4 +136,117 @@ jobs:
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: rocksdb-v18
```
Comment (kwek20) [inline .github/workflows/java_binding_publish.yml:161]: We need this branch on iotaledger repo, or at least it tells me it was "renamed to android" so we should be accurate here

Comment (samuel-rufi) [reply .github/workflows/java_binding_publish.yml:161]: fixed

**Review by kwek20 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**